### PR TITLE
Add bal collector

### PR DESCRIPTION
**Pull request by Goala**
closes https://github.com/abap-observability-tools/abap-metrics-provider/issues/8

### CODE REVIEW

File: src/zamp_collector/zcl_amp_c_bal_logs.clas.abap [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    metrics_current_run = zcl_amp_collector_utils=>initialize_metrics( metrics_last_run = metrics_last_run ).
+
+    SELECT
+    COUNT(*) AS count,
+    object AS object,
+    subobject AS subobject
+    FROM balhdr
+    INTO TABLE @DATA(bal_logs)
+    WHERE aldate = @sy-datum
```
Comment (JohannesKonings) [inline src/zamp_collector/zcl_amp_c_bal_logs.clas.abap:27]: Could you use the date from the framework? 
That comes from a timestamp. If I do not mix up anything sy-datum has the OS date and the timestamp from the application server.

**Review by JohannesKonings — CHANGES_REQUESTED** [review]

File: src/zamp_collector/zcl_amp_c_bal_logs.clas.abap (hunk 1 above)
Comment (Goala) [reply src/zamp_collector/zcl_amp_c_bal_logs.clas.abap]: yes i can
i also reviewed some older collectors and changed it to the frameworks date

Comment (Goala) [reply src/zamp_collector/zcl_amp_c_bal_logs.clas.abap]: https://github.com/abap-observability-tools/abap-metrics-provider/pull/99/files#diff-969b6809c4e3cdd2dff3661b6e481c7d328ae0249f2021a22f06cbd38e6fc08fL22

https://github.com/abap-observability-tools/abap-metrics-provider/pull/99/files#diff-68ea295650e00988be4e40a30eae1055593b3835d28736d439fcc48976d4c63bL27

Comment (JohannesKonings) [reply src/zamp_collector/zcl_amp_c_bal_logs.clas.abap]: well paid attention 😄

**Review by JohannesKonings — APPROVED** [review]
LGTM

**Pull request merged**